sound/soc/sof/pm.c

```diff
@@ -165,6 +165,9 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 		return ret;
 	}
 
+	/* Notify clients about core resume */
+	sof_resume_clients(sdev);
```

Comment:
I still don't get why this is necessary now that you provide a state information to the clients.

Or maybe you need to add a FW_SUSPEND/RESUME state and provide it clients which cannot do power management on their own.